example-form/components/FormRegisterPetDetails/FormRegisterPetDetails.tsx

```diff
@@ -135,15 +135,14 @@ export const FormRegisterPetDetails = () => {
 		(stepFormState?: StepFormState) => {
 			setIsSubmittingStep(true);
 
-			if (stepFormState) {
-				setFormState((current) => ({
-					...current,
-					[currentStep]: { ...stepFormState, completed: true },
-				}));
-			}
-
 			// Using a `setTimeout` to replicate a call to a back-end API
 			setTimeout(() => {
+				if (stepFormState) {
```

Comment:
This fixes a bug that I noticed after looking at this app with fresh eyes.

This will ensure the state for moving to the next step happens _after_ the fake network request finishes.